Services/Certificate/classes/Form/Repository/class.ilCertificateSettingsTestFormRepository.php

```diff
@@ -1,38 +1,20 @@
-<?php
+<?php declare(strict_types=1);
+
 /* Copyright (c) 1998-2018 ILIAS open source, Extended GPL, see docs/LICENSE */
 
+use ILIAS\Filesystem\Exception\FileAlreadyExistsException;
+use ILIAS\Filesystem\Exception\IOException;
+use ILIAS\Filesystem\Exception\FileNotFoundException;
+
 /**
  * @author  Niels Theen <ntheen@databay.de>
  */
 class ilCertificateSettingsTestFormRepository implements ilCertificateFormRepository
 {
-    /**
-     * @var ilCertificateSettingsFormRepository
-     */
-    private $settingsFromFactory;
-
-    /**
-     * @var ilLanguage
-     */
-    private $language;
+    private ?ilCertificateSettingsFormRepository $settingsFromFactory;
```

Comment:
This is not optional. Only the passed constructor argument is optional, the value stored in `$this->settingsFromFactory` MUST never be `null`.